### PR TITLE
mkosi: Rework kconfig handling

### PR DESCRIPTION
**Pull request by daandemeyer**
Instead of relying on the arch specific kconfigs that enable all kinds of options that are irrelevant for our use case, let's beef up our own config a little so that we don't need to rely on the arch specific configs anymore.

Also, instead of using scripts/config to enable options in the mkosi build script, let's add a new kernel.config file that includes all the configs we want enabled/disabled.

Finally, let's make sure we can properly run the bpf selftests by making sure we enable all the right configs required to run the bpf selftests successfully. We use the scripts/kconfig/merge_config.sh script to merge the three different config files together.

### CODE REVIEW

**Comment by bluca** [comment]
Can the kernel config be stored in mkosi.conf.d/ to avoid cluttering the root? Or if not make it a hidden file maybe?

File: kernel.config [hunk 1]
```diff
@@ -0,0 +1,211 @@
+# CONFIG_COMPAT_BRK is not set
+# CONFIG_LEGACY_PTYS is not set
+CONFIG_9P_FS=y
```
Comment (poettering) [inline kernel.config]: hmm?

File: kernel.config [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+CONFIG_CONNECTOR=y
+CONFIG_CPUSETS=y
+CONFIG_CRASH_DUMP=y
+CONFIG_DEBUG_BOOT_PARAMS=y
+CONFIG_DEBUG_FS=y
+CONFIG_DEBUG_INFO_BTF=y
+CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_DEVTMPFS=y
+CONFIG_DM_MIRROR=y
```
Comment (poettering) [inline kernel.config]: hmm?

File: kernel.config [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG_SECONDARY_KEYRING=y
+CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG=y
+CONFIG_DM_VERITY=y
+CONFIG_DM_ZERO=y
```
Comment (poettering) [inline kernel.config]: hmm?

File: kernel.config [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+CONFIG_VIRTIO_NET=y
+CONFIG_VIRTIO_PCI=y
+CONFIG_WATCHDOG=y
+CONFIG_X86_ACPI_CPUFREQ=y
+CONFIG_X86_CPUID=y
+CONFIG_X86_MSR=y
+CONFIG_XFRM_USER=y
+CONFIG_XFS_FS=y
+CONFIG_XFS_POSIX_ACL=y
+CONFIG_YENTA=y
```
Comment (poettering) [inline kernel.config]: hmm?

File: kernel.config [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_DEFAULT="utf8"
+CONFIG_NLS_ISO8859_1=y
+CONFIG_NLS_UTF8=y
+CONFIG_NO_HZ_FULL=y
+CONFIG_NUMA=y
+CONFIG_NVRAM=y
+CONFIG_PACKET=y
+CONFIG_PARAVIRT=y
+CONFIG_PCCARD=y
```
Comment (poettering) [inline kernel.config]: hmm?

File: kernel.config [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+CONFIG_NETLABEL=y
+CONFIG_NF_CONNTRACK_FTP=y
+CONFIG_NF_CONNTRACK_IRC=y
+CONFIG_NF_CONNTRACK_SIP=y
+CONFIG_NF_CONNTRACK=y
+CONFIG_NF_CT_NETLINK=y
+CONFIG_NF_NAT=y
+CONFIG_NFS_FS=y
+CONFIG_NFS_V3_ACL=y
+CONFIG_NFS_V4=y
```
Comment (poettering) [inline kernel.config]: hmm?

**Comment by poettering** [comment]
(only had a superficial look at the options, but some seem surprising to me?)

**Review by bluca — APPROVED** [review]

**Pull request merged**